### PR TITLE
[3.10] Make sure the extension install works as expected

### PR DESCRIPTION
**Pull request by zero-24**
### Summary of Changes

Only show a message when the verification fails (request by @mbabker )
Remove the check at install time.
Make sure we are using the correct update object and don't do some magic with passing the update xml.

### Testing Instructions

- Install 3.9-dev https://developer.joomla.org/nightlies/Joomla_3.9.0-dev-Development-Full_Package.zip
- Install any outdated extension that do not support hashes yet (well all 3rd party tools)
- notice the message saying that there is no hash provided
- install the update
- notice the message saying that there is no hash provided
- apply this patch
- redo the steps above
- notice that there is no message at all.
- install com_patchtester
- notice there is no message
- apply the update to the latest com_patchtester
- notice there is no message at all. (even as they use hashes)

### Expected result

Only show a message when the verification fails (request by @mbabker )
Remove the check at install time.
Make sure we are using the correct update object and don't do some magic with passing the update xml.

### Documentation Changes Required

None.

### CODE REVIEW

**Comment by zero-24** [comment]
Tagging @alikon too ;)

File: administrator/components/com_installer/models/update.php [hunk 1]
```diff
@@ -443,26 +443,19 @@ private function install($update, $updateurl)
...
 		// Get an installer instance
 		$installer = JInstaller::getInstance();
 		$update->set('type', $package['type']);
 
 		// Check the package
-		$check = JInstallerHelper::isChecksumValid($package['packagefile'], (string) $updateurl);
+		$check = InstallerHelper::isChecksumValid($package['packagefile'], $update);
 
-		switch ($check)
+		// The validation was not successful for now just a warning for now.
```
Comment (brianteeman) [inline administrator/components/com_installer/models/update.php]: The validation was not successful. Just a warning for now.

**Review by brianteeman — CHANGES_REQUESTED** [review]

**Review by brianteeman — APPROVED** [review]

File: libraries/src/Installer/InstallerHelper.php [hunk 2]
```diff
@@ -361,31 +361,28 @@ public static function splitSql($query)
 	/**
 	 * Return the result of the checksum of a package with the SHA256/SHA384/SHA512 tags in the update server manifest
 	 *
-	 * @param   string     $packagefile           Location of the package to be installed
-	 * @param   Installer  $updateServerManifest  Update Server manifest
+	 * @param   string   $packagefile   Location of the package to be installed
+	 * @param   JUpdate  $updateObject  The Update Object
 	 *
-	 * @return  integer  one if the hashes match, zero if hashes doesn't match, two if hashes not found
+	 * @return  boolean    False in case the validation did not work; true in any other case.
```
Comment (Quy) [inline libraries/src/Installer/InstallerHelper.php]: Shouldn't it be `integer`?

Comment (zero-24) [reply libraries/src/Installer/InstallerHelper.php]: thnaks the initial plan was to change the method but i have just reverted that change.

File: administrator/components/com_installer/models/update.php [hunk 3]
```diff
@@ -443,26 +443,19 @@ private function install($update, $updateurl)
...
 		// Get an installer instance
 		$installer = JInstaller::getInstance();
 		$update->set('type', $package['type']);
 
 		// Check the package
-		$check = JInstallerHelper::isChecksumValid($package['packagefile'], (string) $updateurl);
+		$check = InstallerHelper::isChecksumValid($package['packagefile'], $update);
 
-		switch ($check)
+		// The validation was not successful. Just a warning for now.
```
Comment (mbabker) [inline administrator/components/com_installer/models/update.php:455]: I'd add a TODO here saying that we'll basically restore this switch case in 4.0.  Beyond that PR's good by me.

File: administrator/language/en-GB/en-GB.com_installer.ini [hunk 4]
```diff
@@ -64,9 +64,7 @@ COM_INSTALLER_HEADING_UPDATESITE_NAME_ASC="Update Site ascending"
 COM_INSTALLER_HEADING_UPDATESITE_NAME_DESC="Update Site descending"
 COM_INSTALLER_HEADING_UPDATESITEID="ID"
 COM_INSTALLER_INSTALL_BUTTON="Install"
-COM_INSTALLER_INSTALL_CHECKSUM_CORRECT="File Checksum OK"
-COM_INSTALLER_INSTALL_CHECKSUM_NOT_FOUND="There were no checksums provided in the package."
-COM_INSTALLER_INSTALL_CHECKSUM_WRONG="File Checksum Failed"
+COM_INSTALLER_INSTALL_CHECKSUM_WRONG="The checksum verification failed. Please make sure you are using the correct update server!"
```
Comment (Quy) [inline administrator/language/en-GB/en-GB.com_installer.ini:67]: For the second sentence, how about `Be sure to use the correct update server!`?

Comment (brianteeman) [reply administrator/language/en-GB/en-GB.com_installer.ini:67]: The current string is fine and is simpler language than the suggestion.

I missed that you have a space at the end before the exclamation mark that should be removed

File: libraries/src/Installer/InstallerHelper.php [hunk 5]
```diff
@@ -361,31 +361,28 @@ public static function splitSql($query)
...
-	 * @param   Installer  $updateServerManifest  Update Server manifest
+	 * @param   string   $packagefile   Location of the package to be installed
+	 * @param   JUpdate  $updateObject  The Update Object
 	 *
 	 * @return  integer  one if the hashes match, zero if hashes doesn't match, two if hashes not found
 	 *
 	 * @since   __DEPLOY_VERSION__
 	 */
-	public static function isChecksumValid($packagefile, $updateServerManifest)
+	private function isChecksumValid($packagefile, $updateObject)
```
Comment (wilsonge) [inline libraries/src/Installer/InstallerHelper.php]: still public static :P

**Review by wilsonge — CHANGES_REQUESTED** [review]

**Review by wilsonge — APPROVED** [review]

**Comment by alikon** [comment]
Why remove the check at install time ?

**Comment by infograf768** [comment]
Languages install/reinstall/uninstall fine with this patch.

File: administrator/language/en-GB/en-GB.com_installer.ini (hunk 4 above)
Comment (zero-24) [reply administrator/language/en-GB/en-GB.com_installer.ini:67]: > I missed that you have a space at the end before the exclamation mark that should be removed

Was this a suggestion for me? As i don't see a space at the end before the exclamation mark?

File: administrator/components/com_installer/models/update.php (hunk 3 above)
Comment (zero-24) [reply administrator/components/com_installer/models/update.php:455]: George committed the todo just one line below so this is fine ;)

File: administrator/language/en-GB/en-GB.com_installer.ini (hunk 4 above)
Comment (brianteeman) [reply administrator/language/en-GB/en-GB.com_installer.ini:67]: I dont either - I must have been smoking something - all good for me

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on a6059d5a0462266d4c5d396a9753624b9c5e513a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20330">issues.joomla.org/tracker/joomla-cms/20330</a>.</sub>

**Comment by brianteeman** [comment]
(updated title to show it is now for 3.10)

**Comment by zero-24** [comment]
Moved: https://github.com/joomla/joomla-cms/pull/20646

**Pull request closed without merging**